internal/providers/openstack/openstack.go

```diff
@@ -50,7 +51,18 @@ var (
 	}
 )
 
-func FetchConfig(f *resource.Fetcher) (types.Config, report.Report, error) {
+func init() {
+	platform.Register(platform.Provider{
```

Comment:
Why is openstack special? it seems to have two platforms registered?

Reply:
This is a bit of awkwardness around the new model.  In the old model, all the registrations were in one place, and were technically able to mix and match functions from different provider backends.  That happened in two places: `brightbox` was registered using the `openstack` fetch function, and `metal` was registered as the only user of the `noop` provider.

I think the new structure is clearer for the common case, but we need to deal with those oddities.  For `noop` I simply renamed the provider to `metal`.  For `brightbox` I could have copypasted the entire `openstack` provider, but that seems like unnecessary code duplication, so I just added a second registration inside the `openstack` provider.  It's a bit obscure, but seems like it's probably fine?

Reply:
Ah ok, yeah that makes sense. thank you for the feedback. It felt weird from a 100ft view but now that its explained I think that makes sense. Could we maybe document it a little in a comment?

Reply:
Yup, done!